mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/SmsNotificationServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+package org.motechproject.nms.mobileacademy.service.impl;
+
+import org.apache.commons.httpclient.HttpStatus;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.joda.time.DateTime;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.event.listener.annotations.MotechListener;
+import org.motechproject.nms.mobileacademy.domain.CompletionRecord;
+import org.motechproject.nms.mobileacademy.repository.CompletionRecordDataService;
+import org.motechproject.nms.mobileacademy.service.SmsNotificationService;
+import org.motechproject.server.config.SettingsFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * This handles all the integration pieces between MA and sms module to trigger and handle notifications
+ * for course completion
+ */
+@Service("smsNotificationService")
+public class SmsNotificationServiceImpl implements SmsNotificationService {
+
+    private static final String COURSE_COMPLETED = "nms.ma.course.completed";
+
+    private static final String SMS_STATUS = "nms.ma.sms.deliveryStatus";
```

Comment:
Possible, yes. But, then it would live in a common module or the controller would need to take a reference on the service class with the constant defined. I'm not sure if I like either design, but I can do whatever is recommended. Just adding the subject suffix for now.